cpp/src/lance/encodings/plain.cc

```diff
@@ -37,14 +37,37 @@ ::arrow::Result<int64_t> PlainEncoder::Write(std::shared_ptr<::arrow::Array> arr
   ARROW_ASSIGN_OR_RAISE(auto value_offset, out_->Tell());
   // TODO: support more types.
```

Comment:
Do stuff like the half float, decimal, date time, etc arrow types require special handling? Would it make sense to special case time series data to use RLE?